crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,18 +83,30 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    let line_length: u16 = checker.settings.line_length.into();
+                    if !locator.contains_line_break(source_range)
+                        && locator.full_line(source_range.start()).trim_end().len()
+                            <= ((line_length as usize) - 2)
```

Comment:
I would suggest using `usize::from(line_length)` here. They are equivalent in this context because a `u16` will always fit into a `usize`, but `as` casts are something of an annoyance in Rust-land. (Because in general they do silent truncation. And... other things.)

Also, I'm not sure what the acceptable bounds of `checker.settings.line_length` are. IIRC, its minimum value is `1`, so subtracing `2` could result in an underflow. For that, I think I'd suggest saturating arithmetic.

So, that's `usize::from(line_length).saturating_sub(2)`.

(Although I now see Micha's comment, so this code might need to change to be a little more complicated.)

Reply:
Very good point about the underflow, though! I forgot about that.